charts/backstage/README.md

```diff
@@ -240,6 +236,39 @@ Now that the ConfigMap has been created on your Kubernetes cluster, you can refe
 
 The chart will mount the content of the ConfigMap as a new `app-config.extra.yaml` file and automatically pass the extra configuration to your instance.
 
+It's also possible to add your configuration to the helm values.yaml and create the ConfigMap as part of your helm install.
+
+```yaml
+  extraAppConfig:
+    - filename: app-config.yaml
+      configMapRef: app-config
+      data: |
+        app:
+          title: Scaffolded Backstage App
+          baseUrl: http://localhost:3000
+```
+
+This allows multiple app configs to be added. i.e.
+
+```yaml
+  extraAppConfig:
+    - filename: app-config.yaml
+      configMapRef: app-config
+      data: |
+        app:
+          title: Scaffolded Backstage App
+          baseUrl: http://localhost:3000
+        ...  
+    - filename: app-config.production.yaml
+      configMapRef: app-config-production    
+      data: |
+        app:
+          title: somedomain
+          baseUrl: https://somedomain.tld
```

Comment:
What backstage behavior and use case is supported by this? Configmaps are merged into a single global config object in runtime anyways. I don't see a purpose in inlining multiple config values at the same place. Can you please explain me when this may be helpful?

Reply:
When packaging the helm chart it should be able to work in the local, development and production environments, with just changes to the values files for the environment deployed. The different configurations using for example different database backends sql lite or postgres.

Reply:
I don't understand that. If you define `extraAppConfig` as a list, all the configs are merged, applied and deployed to the backstage instance. There's no way for you to select which config to apply without changing the `values.yaml`, all of them get applied at once.

If you need to deploy to different environments with different values, you need to have different `values.yaml` for each environment.


Can you please share some examples on how do you imagine this to work?

Reply:
What was the intended purpose of the `extraAppConfig` node ? was it not to allow the possibility of allowing multiple files to be mountable and accessible to backstage?

Reply:
> I don't understand that. If you define `extraAppConfig` as a list, all the configs are merged, applied and deployed to the backstage instance. There's no way for you to select which config to apply without changing the `values.yaml`, all of them get applied at once.
> 
> If you need to deploy to different environments with different values, you need to have different `values.yaml` for each environment.
> 
> Can you please share some examples on how do you imagine this to work?

Here's it working https://github.com/polarpoint-io/argocd-tooling-applications/blob/main/releases/apps/platform/backstage/non-prod.yaml